webhook/resources/volume/validator.go

```diff
@@ -156,6 +164,53 @@ func (v *volumeValidator) Update(request *admission.Request, oldObj runtime.Obje
 	return nil
 }
 
+func (v *volumeValidator) validateExpansionSize(oldVolume *longhorn.Volume, newVolume *longhorn.Volume) error {
+	oldSize := oldVolume.Spec.Size
+	newSize := newVolume.Spec.Size
+	if newSize == oldSize {
+		return nil
+	}
+	if newSize < oldSize && !newVolume.Status.ExpansionRequired {
+		return fmt.Errorf("shrinking volume size from %v/%v to %v/%v is not supported", oldVolume.Name, oldSize, newVolume.Name, newSize)
```

Comment:
nit: volume name is the same, so the error message can be simplified.

Reply:
Modified. just concerned about that old and new volume not the same one.
Thanks